### PR TITLE
More efficient query methods, extended the docs a bit

### PR DESCRIPTION
**Pull request by aleksclark**

### CODE REVIEW

File: README.markdown [hunk 1]
```diff
@@ -88,7 +142,7 @@ folder.update
 * Listing the comments in a discussion surrounding a folder.
 
 ```ruby
-folder = client.folder('image_folder')
+folder = client.folder('image_folder') # using a path (seriously, don't do this)
```
Comment (bcoe) [inline README.markdown]: The updates to the API are appreciated, but I don't like the tone of some of these comments. Perhaps just put a comment after the lookups by filename which indicates: # lookups by id are more efficient, and provide an example of lookups by ID (you already do this).

File: lib/ruby-box/client.rb [hunk 2]
```diff
@@ -12,7 +12,11 @@ def initialize(session)
     end
 
     def root_folder
-      folder = Folder.new(@session, {'id' => '0'})
+      folder_by_id('0')
+    end
+
+    def folder_by_id(id)
```
Comment (bcoe) [inline lib/ruby-box/client.rb:18]: Great addition to functionality :thumbsup:

File: lib/ruby-box/folder.rb [hunk 3]
```diff
@@ -77,6 +77,15 @@ def has_mini_format?
       true
     end
 
+    def collection_for(key)
```
Comment (bcoe) [inline lib/ruby-box/folder.rb]: This seems like a great addition as well; you use the list of items already loaded on a file or folder to begin returning results, before querying a page from the API?

Do you mind pulling this functionality in to a separate pull request? I'd like to get integration tests around it.

**Comment by rposborne** [comment]
@aleksclark any progress on @bcoe requests? This is some great work, hate to see it sit in the PR graveyard.

**Comment by aleksclark** [comment]
@rposborne yea graveyard :P thanks for the kick in the pants

**Comment by doliveirakn** [comment]
+1

**Comment by aleksclark** [comment]
(just in case nobody notices I did comply with @bcoe 's request after @rposborne reminded me)

**Pull request merged**